### DISCUSSION TITLE
useFieldArray, length validation related to whole array, TypeScript, V7

### DISCUSSION THREAD

**Discussion by xkrchnav** [react-hook-form/react-hook-form#6316] [Q&A] [answered]
Hello,
We have the following use case pretty often - create an array of fields, and then there is a validation for whole array - at least one item in field, mix/max number of items, some calculation etc.

```typescript
type FormData = {
  inputs: number[]
};

const { register, formState: errors } = useForm<FormData>();

// Type question here:
// type of errors.inputs is (FieldError | undefined)[] | undefined
// but errors.inputs as unknown as FieldError allows to pick .message
...
<input {...register(`inputs.${i}`, { required: "This input is required" })} />
```

So the question is - **what is the proper way to validate length of the fieldArray**? Or possibly min number of items in array etc? Do we need to create additional field in FormData to hold number of inputs? What's the indented usage of FieldError on array?

Thank you,
Jozef

**Comment by bluebill1049** [comment, marked as answer]
https://github.com/react-hook-form/react-hook-form/tree/master/examples

Field Array Min Length

**Reply by xkrchnav** [reply]
Thank you! That was quick :)
Edit: trying to set mark as answer but "bubble" doesn't want to disappear so at least added points... will try later.
